pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller.go

```diff
@@ -0,0 +1,185 @@
+)
+
+// ClusterExternalSecretReconciler reconciles a ClusterExternalSecret object.
+type Reconciler struct {
+	client.Client
+	Log             logr.Logger
+	Scheme          *runtime.Scheme
+	RequeueInterval time.Duration
+}
+
+const (
+	errGetCES              = "could not get ClusterExternalSecret"
+	errPatchStatus         = "unable to patch status"
+	errLabelMap            = "unable to get map from labels"
+	errNamespaces          = "could not get namespaces from selector"
+	errGetExistingES       = "could not get existing ExternalSecret"
+	errCreatingOrUpdating  = "could not create or update ExternalSecret"
+	errSetCtrlReference    = "could not set the controller owner reference"
+	errSecretAlreadyExists = "secret already exists in namespace"
+)
+
+//+kubebuilder:rbac:groups=external-secrets.io,resources=clusterexternalsecrets,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=external-secrets.io,resources=clusterexternalsecrets/status,verbs=get;update;patch
+
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("ClusterExternalSecret", req.NamespacedName)
+
+	var clusterExternalSecret esv1alpha1.ClusterExternalSecret
+
+	err := r.Get(ctx, req.NamespacedName, &clusterExternalSecret)
+	if apierrors.IsNotFound(err) {
+		return ctrl.Result{}, nil
+	} else if err != nil {
+		log.Error(err, errGetCES)
+		return ctrl.Result{}, nil
+	}
+
+	p := client.MergeFrom(clusterExternalSecret.DeepCopy())
+	defer func() {
+		err = r.Status().Patch(ctx, &clusterExternalSecret, p)
+		if err != nil {
+			log.Error(err, errPatchStatus)
+		}
+	}()
+
+	// Fetch Namespaces to grab ExternalSecrets
+	genClient := kubernetes.NewForConfigOrDie(ctrl.GetConfigOrDie())
+	namespaces := genClient.CoreV1().Namespaces()
+
+	refreshInt := r.RequeueInterval
+	if clusterExternalSecret.Spec.RefreshInterval != nil {
+		refreshInt = clusterExternalSecret.Spec.RefreshInterval.Duration
+	}
+
+	labelMap, err := metav1.LabelSelectorAsMap(&clusterExternalSecret.Spec.NamespaceSelector)
+	if err != nil {
+		log.Error(err, errLabelMap)
+		return ctrl.Result{RequeueAfter: refreshInt}, nil
+	}
+
+	namespaceList, err := namespaces.List(ctx, metav1.ListOptions{LabelSelector: labels.SelectorFromSet(labelMap).String()})
+	if err != nil {
+		log.Error(err, errNamespaces)
+		return ctrl.Result{RequeueAfter: refreshInt}, nil
+	}
+
+	esName := clusterExternalSecret.Spec.ExternalSecretName
+	if esName == "" {
+		esName = clusterExternalSecret.ObjectMeta.Name
+	}
+
+	failedNamespaces := []string{}
+
+	for _, namespace := range namespaceList.Items {
+		var existingES esv1alpha1.ExternalSecret
+		err = r.Get(ctx, types.NamespacedName{
+			Name:      esName,
+			Namespace: namespace.Name,
+		}, &existingES)
+
+		if err != nil && !apierrors.IsNotFound(err) {
+			log.Error(err, errGetExistingES)
+		}
+
+		// No one owns this resource so error out
+		if len(existingES.ObjectMeta.OwnerReferences) == 0 {
```

Comment:
```suggestion
		if !apierrors.IsNotFound(err) && en(existingES.ObjectMeta.OwnerReferences) == 0 {
```
This is because the default empty ExternalSecrets object has no ownerReferences, so it is always zero.